spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CAddTable.scala

```diff
@@ -38,6 +41,42 @@ class CAddTable[T: ClassTag, D: ClassTag](val inplace: Boolean = false)(
 
   output = Tensor[D]()
 
+  @transient
+  private var bufferSumInput: Tensor[D] = null
+  @transient
+  private var bufferSumOutput: Tensor[D] = null
+
+  private def canBroadcast(tensor: Tensor[D], other: Tensor[D]): Boolean = {
```

Comment:
from the implementation,  we don't really use `tensor` we just use `size()` instead, probably use `size: Array[Int]` as first parameter and then make it a more meaningful name

Reply:
How about ` private def dimSumResizable(tensor: Tensor[D],  size: Array[Int]): Boolean = {...}`,  or a better one

Reply:
OK, how about change to 'dimResizable'?

Reply:
I think the intention of this method is to sum along specific `dim` and then check if the summed tensor can be resized (identical) to referred tensor ?   that's why I said `dimSum`, we can think a better one

Reply:
If two tensors have same dims but different size, size(dim) for small tensor should be 1 or same with another. Then in backward, sum gradOutput along dim that size(dim) is 1.

So firstly we need to check 'dimResizable' (or other name),  and then do `sumAlongDims`.

Reply:
change to `expandWithDims`